examples/hillas_analysis.py

```diff
@@ -0,0 +1,73 @@
+'''
+This example shows how you should use ctapipe.core.tool.Tool
+for your analysis as example for a simple Hillas based analysis.
+'''
+from ctapipe.core.tool import Tool
+from ctapipe.core.traits import Unicode, Dict, List
+from ctapipe.utils import get_dataset
+from ctapipe.io.eventsourcefactory import EventSourceFactory
+from ctapipe.calib import CameraCalibrator
+from ctapipe.image.hillas import hillas_parameters
+from ctapipe.image.cleaning import tailcuts_clean
+
+import astropy.units as u
+
+import json
+
+
+class HillasAnalysis(Tool):
+    # class members are options, that can be set via commandline or config file
+
+    # inputfile, if None we will use the gamma test file
+    inputfile = Unicode(None, allow_none=True).tag(config=True)
+
+    # outputfile
+    outputfile = Unicode('', allow_none=False).tag(config=True)
+
+    classes = List([CameraCalibrator, EventSourceFactory])
+
+    aliases = Dict({'outputfile': 'HillasAnalysis.outputfile'})
+
+    def setup(self):
+        print('inputfile', self.inputfile)
+        print('outputfile', self.outputfile)
+        self.event_source = EventSourceFactory.produce(
+            tool=self, config=self.config,
+            input_url=self.inputfile or get_dataset('gamma_test.simtel.gz')
+        )
+        self.calibrator = CameraCalibrator(
```

Comment:
this should probably explicitly use a set of `Camera*Factory` classes rather than the `CameraCalibrator`, so all the various algorithm options are available. Ideally we should make that a simpler process.

Reply:
Can you give some more hints how to uses those?